source/config/__init__.py

```diff
@@ -1304,6 +1305,56 @@ def __setitem__(
 		self.manager._markWriteProfileDirty()
 		self._cache[key] = val
 
+		# Alias [documentFormatting][reportFontAttributes] for backwards compatibility.
+		# Todo: Remove in 2025.1.
+		if version_year < 2025 and NVDAState._allowDeprecatedAPI():
+			self._linkDeprecatedValues(key, val)
+
+	def _linkDeprecatedValues(self, key: aggregatedSection._cacheKeyT, val: aggregatedSection._cacheValueT):
+		"""Link deprecated config keys and values to their replacements.
+
+		Args:
+			key: The configuration key to link to its new or old counterpart.
+			val: The value associated with the configuration key.
+
+		postconditions:
+			- If self.path is "documentFormatting":
+				- If key is "reportFontAttributes":
+					- If val is True, "documentFormatting.fontAttributeReporting" is set to OutputMode.SPEECH_AND_BRAILLE, otherwise, it is set to OutputMode.OFF.
+				- If key is "fontAttributeReporting":
+					- if val is OutputMode.OFF, "documentFormatting.reportFontAttributes" is set to False, otherwise, it is set to True.
+		"""
+		match self.path:
+			case ("documentFormatting",):
+				match key:
+					case "fontAttributeReporting":
+						# Alias documentFormatting.fontAttributeReporting to documentFormatting.reportFontAttributes for backwards compatibility.
+						key = "reportFontAttributes"
+						val = bool(val)
+
+					case "reportFontAttributes":
+						# Alias documentFormatting.reportFontAttributes to documentFormatting.fontAttributeReporting for forwards compatibility.
+						log.warning(
+							"documentFormatting.reportFontAttributes is deprecated. Use documentFormatting.fontAttributeReporting instead.",
+							# Include stack info so testers can report warning to add-on author.
+							stack_info=True,
+						)
+						key = "fontAttributeReporting"
+						val = OutputMode.SPEECH_AND_BRAILLE if val else OutputMode.OFF
+
+					case _:
+						# We don't care about other keys in this section.
+						return
+
+			case _:
+				# We don't care about other sections.
+				return
+
+		# Update the value in the most recently activated profile.
```

Comment:
Have you considered overriding __get_item as well? That would allow you to:
1. Remove `reportFontAttributes` from the profile
2. When getting `reportFontAttributes`, log a warning and get the bool value of `fontAttributeReporting` instead
3. When setting `reportFontAttributes`, log a warning and set `fontAttributeReporting` instead
4. Do nothing special when getting or setting `fontAttributeReporting`

Reply:
I did think of this, and started implementing it. However, the logic for these cases proved to be a little fiddly to get working well. Now, that may have just been me, or even just me last week, but I think the current solution works fine, and adds pretty minimal complexity, which means fewer things to go wrong, and less technical debt. If you can think of a reason why the current approach won't work, I'm happy to hear it and consider adding a case to the getter as well.

Reply:
AH, I didn't know you investigated this route. If it turned out to be too complex, I'm perfeclty fine with the current approach.